pkg/test-fw/ig/command.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ig
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type Command struct {
+	// IG wrapper
+	IG
+
+	// Name of the command to be run, used to give information.
+	Name string
+
+	// ExpectedString contains the exact expected output of the command.
+	ExpectedString string
+
+	// ExpectedRegexp contains a regex used to match against the command output.
+	ExpectedRegexp string
```

Comment:
I think we can get rid of these two things. `ValidateOutput` is generic enough and should be the only way we provide. If an user wants to validate against an exact string or regexp they can do something like:

```go
ValidateOutput: func(t *testing.T, output string) {
			require.Equal(t, "foo", output)
		},
```

We can even provide a helper like this:

```go
func MatchStringFunc(t *testing.T, expected string) func(t *testing.T, output string) {
	return func(t *testing.T, output string) {
		require.Equal(t, expected, output)
	}
}
```

to be used like:
```go
ValidateOutput: MatchStringFunc(t, "foo"),
```

(Same idea applies to the regexp case.

Reply:
@mauriciovasquezbernal Should we fail the test if ValidateOutput func is nil (not specified by the user), or simply return a log statement?

Reply:
I think nothing special should happen in that case. It's possible that users only want to run a command without worrying about their output.